test/libsolidity/util/ContractABIUtils.cpp

```diff
@@ -33,55 +39,256 @@ using namespace dev::solidity::test;
 using namespace std;
 using namespace soltest;
 
-dev::solidity::test::ParameterList ContractABIUtils::parametersFromJson(
+namespace
+{
+
+using ParameterList = dev::solidity::test::ParameterList;
+
+auto arraySize(string const& _arrayType) -> size_t
+{
+	auto leftBrack = _arrayType.find("[");
+	auto rightBrack = _arrayType.find("]");
+
+	soltestAssert(leftBrack != string::npos && rightBrack != string::npos, "");
+
+	string size = _arrayType.substr(leftBrack + 1, rightBrack - leftBrack - 1);
+	return static_cast<size_t>(stoi(size));
+}
+
+auto isBool(string _type) -> bool
+{
+	return regex_match(_type, regex{"(bool)"});
+}
+
+auto isUint(string _type) -> bool
+{
+	return regex_match(_type, regex{"(uint\\d*)"});
+}
+
+auto isInt(string _type) -> bool
+{
+	return regex_match(_type, regex{"(int\\d*)"});
+}
+
+auto isBytes(string _type) -> bool
+{
+	return regex_match(_type, regex{"(bytes\\d+)"});
+}
+
+auto isDynBytes(string _type) -> bool
+{
+	return regex_match(_type, regex{"(\\bbytes\\b)"});
+}
+
+auto isString(string _type) -> bool
+{
+	return regex_match(_type, regex{"(string)"});
+}
+
+auto isBoolArray(string _type) -> bool
+{
+	return regex_match(_type, regex{"(bool)(\\[\\d+\\])"});
+}
+
+auto isUintArray(string _type) -> bool
+{
+	return regex_match(_type, regex{"(uint\\d*)(\\[\\d+\\])"});
+}
+
+auto isIntArray(string _type) -> bool
+{
+	return regex_match(_type, regex{"(int\\d*)(\\[\\d+\\])"});
+}
+
+auto isStringArray(string _type) -> bool
+{
+	return regex_match(_type, regex{"(string)(\\[\\d+\\])"});
+}
+
+auto isTuple(string _type) -> bool
+{
+	return regex_match(_type, regex{"(tuple)"});
+}
+
+auto isTupleArray(string _type) -> bool
+{
+	return regex_match(_type, regex{"(tuple)(\\[\\d+\\])"});
+}
+
+auto functionMatchesSignature(Json::Value const& _contractABIFunction, string const& _signature) -> bool
+{
+	string functionName{_signature.substr(0, _signature.find("("))};
```

Comment:
Maybe better to just construct the signature from _contractABIFunction and then check for string equality?

Reply:
Yes, that's a lot clearer. Done.